### PR TITLE
Bump rebound/reboundx to version 4.4.3/4.3.0

### PR DESCRIPTION
**Pull request by hannorein**
### Description

Bumps the REBOUND version from 3.24.2 to 4.4.3. Also bumps REBOUNDx to 4.3.0 Fixes the compile time error mentioned in #4925.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry

### CODE REVIEW

File: packages/rebound/patches/0001-fix-install_name.patch
Comment (agriyakhetarpal) [inline packages/rebound/patches/0001-fix-install_name.patch:1]: Could this patch be dropped – I noticed that the build worked fine without it?

File: packages/rebound/patches/0002-fix-output.patch
Comment (agriyakhetarpal) [inline packages/rebound/patches/0002-fix-output.patch:1]: If the first patch is dropped, could you rename the file with "0001" instead?

**Review by agriyakhetarpal — APPROVED** [review]
Thanks, @hannorein! Just a minor nitpick-y suggestion.

**Comment by hannorein** [comment]
It gets rid of a warning message so I thought I'll keep it for now. I'll merge these patches into the upstream in a future version of REBOUND so that we can get rid of all them.

**Comment by agriyakhetarpal** [comment]
There's another error: `Error: Dynamic linking error: cannot resolve symbol emscripten_sleep` :(

**Comment by hannorein** [comment]
I can fix that. But how come I didn't get those errors? And how can I make sure I fix them all?

**Comment by agriyakhetarpal** [comment]
> I can fix that. But how come I didn't get those errors? And how can I make sure I fix them all?

You could run the tests locally with `pytest -svra packages/rebound/test_rebound.py` – but I usually find the CI logs to be better  sometimes, because from a local run I wasn't able to see that the timeout was due to a Pyodide fatal error (this one, in this case).

I'd recommend an out-of-tree build in your CI, too, if that helps and if you're interested: https://pyodide.org/en/stable/development/building-and-testing-packages.html

**Comment by hannorein** [comment]
Thanks. I'm still a bit confused on how to do this properly. 

Right now the tests pass locally. If the package compiles and runs fine locally, then why does it not compile in the CI? 

Are the two failing CI tests `main / test-core (macos-14, selenium, safari)` and `ci/circleci: test-core-node-nodylink` failing because of something in the rebound/reboundx packages? Or is that unrelated?

**Comment by hoodmane** [comment]
Those are unrelated, don't worry about them.

**Comment by hoodmane** [comment]
CI looks good to me.

**Comment by hoodmane** [comment]
Thanks @hannorein!

**Pull request merged**